============================= test session starts ==============================
platform linux -- Python 3.9.0, pytest-6.1.2, py-1.9.0, pluggy-0.13.1
rootdir: /home/chips/aulas/compiladores/trabalhos/lispy-parser/run/180117548
plugins: json-report-1.2.4, metadata-1.11.0, hypothesis-5.41.4
collected 12 items

test_parser.py .........FFF                                              [100%]

=================================== FAILURES ===================================
_________________ TestAnalisadorSintatico.test_converte_chars __________________

self = <test_parser.TestAnalisadorSintatico object at 0x7f257a511d00>

    def test_converte_chars(self):
>       assert value(r"#\A") == "A"
E       AssertionError: assert Tree('char', [Token('CHAR', '#\\A')]) == 'A'
E        +  where Tree('char', [Token('CHAR', '#\\A')]) = value('#\\A')

test_parser.py:122: AssertionError
__ TestAnalisadorSintatico.test_inclui_comando_begin_em_sequencia_de_comandos __

self = <test_parser.TestAnalisadorSintatico object at 0x7f257a57aac0>

    def test_inclui_comando_begin_em_sequencia_de_comandos(self):
>       assert value("(cmd 1)\n(cmd 2)") == [
            symbol("begin"),
            [symbol("cmd"), 1],
            [symbol("cmd"), 2],
        ]
E       assert "No terminal defined for '(' at line 2 col 1\n\n(cmd 2)\n^\n" == [Symbol(value='begin'), [Symbol(value='cmd'), 1], [Symbol(value='cmd'), 2]]
E        +  where "No terminal defined for '(' at line 2 col 1\n\n(cmd 2)\n^\n" = value('(cmd 1)\n(cmd 2)')

test_parser.py:127: AssertionError
_____ TestAnalisadorSintatico.test_aceita_programa_completo_com_comentário _____

self = <test_parser.TestAnalisadorSintatico object at 0x7f257a452ca0>

    def test_aceita_programa_completo_com_comentário(self):
        src = """
        ;; Fatorial
        (define fat (lambda (n)
            (if (<= n 1)
                1
                (* n (fat (- n 1))))))
    
        (print (fat 5))
        """
        print("Testando comando")
        print(src)
        s = symbol
>       assert value(src) == [
            s("begin"),
            [
                s("define"),
                s("fat"),
                [
                    s("lambda"),
                    [s("n")],
                    [
                        s("if"),
                        [s("<="), s("n"), 1],
                        1,
                        [s("*"), s("n"), [s("fat"), [s("-"), s("n"), 1]]],
                    ],
                ],
            ],
            [s("print"), [s("fat"), 5]],
        ]
E       assert "No terminal defined for '(' at line 8 col 9\n\n        (print (fat 5))\n        ^\n" == [Symbol(value='begin'), [Symbol(value='define'), Symbol(value='fat'), [Symbol(value='lambda'), [Symbol(value='n')], [S...bol(value='*'), Symbol(value='n'), [Symbol(value='fat'), [...]]]]]], [Symbol(value='print'), [Symbol(value='fat'), 5]]]
E        +  where "No terminal defined for '(' at line 8 col 9\n\n        (print (fat 5))\n        ^\n" = value('\n        ;; Fatorial\n        (define fat (lambda (n) \n            (if (<= n 1)\n                1\n                (* n (fat (- n 1))))))\n\n        (print (fat 5))\n        ')

test_parser.py:147: AssertionError
----------------------------- Captured stdout call -----------------------------
Testando comando

        ;; Fatorial
        (define fat (lambda (n) 
            (if (<= n 1)
                1
                (* n (fat (- n 1))))))

        (print (fat 5))
        
--------------------------------- JSON report ----------------------------------
JSON report written to: .report.json (12324 bytes)
=========================== short test summary info ============================
FAILED test_parser.py::TestAnalisadorSintatico::test_converte_chars - Asserti...
FAILED test_parser.py::TestAnalisadorSintatico::test_inclui_comando_begin_em_sequencia_de_comandos
FAILED test_parser.py::TestAnalisadorSintatico::test_aceita_programa_completo_com_comentário
========================= 3 failed, 9 passed in 0.18s ==========================
